### PR TITLE
Define `blsKeyAndBftWeightSchema` and inject it where used

### PR DESCRIPTION
**Pull request by mjerkov**
Resolves Issue [#290](https://github.com/LiskHQ/lips/issues/290).

In LIP 0058, we define a new schema `blsKeyAndBftWeightSchema`, and then inject it where it occurs:
- In LIP 0058: `validatorsHashInputSchema`
- In LIP 0045: `chainValidatorsSchema`
- In LIP 0043: `sidechainRegParams`, `mainchainRegParams`, and `registrationSignatureMessageSchema`.

### CODE REVIEW

File: proposals/lip-0058.md [hunk 1]
```diff
@@ -364,30 +364,41 @@ The value of the validators hash corresponding to the validator information and
 
 ##### Execution
 
-We use the following JSON schema for the computation of the validators hash.
+We define JSON schema for validators' BLS keys and BFT weights:
+
+```java
+validatorsBlsKeysAndBftWeightsSchema = {
```
Comment (ricott1) [inline proposals/lip-0058.md:370]: this is not a valid Lisk JSON schema, as the root property is not of type object. I would prefer to define the schema for the validator 

```
"type": "object",
"required": ["blsKey", "bftWeight"],
"properties": {
    "blsKey": {
        "dataType": "bytes",
        "length": BLS_PUBLIC_KEY_LENGTH,
        "fieldNumber": 1
    },
    "bftWeight": {
         "dataType": "uint64",
          "fieldNumber": 2
    }
}
```

and keep the array part in the original schema

**Review by ricott1 — CHANGES_REQUESTED** [review]
I would do this differently, as indicated in a comment. I would only define valid Lisk JSON schemas (type==object, see point 1 here: https://github.com/LiskHQ/lips/blob/main/proposals/lip-0027.md#lisk-json-schemas) and keep the array in the original schema.

File: proposals/lip-0058.md (hunk 1 above)
Comment (mjerkov) [reply proposals/lip-0058.md]: Agreed. It was a bit strange to me why this schema doesn't start with the type object, now I see why. Thanks!

File: proposals/lip-0043.md [hunk 2]
```diff
@@ -148,25 +148,9 @@ sidechainRegParams = {
             "fieldNumber": 2
         },
         "sidechainValidators": {
-            "type": "array",
-            "minItems": 1,
+            ...validatorsBlsKeysAndBftWeightsSchema,
```
[the remark this answers is not in the document]
Comment (mjerkov) [reply proposals/lip-0043.md]: Agreed.

File: proposals/lip-0043.md [hunk 3]
```diff
@@ -149,24 +149,10 @@ sidechainRegParams = {
         },
         "sidechainValidators": {
             "type": "array",
+            "items": ...blsKeyAndBftWeightSchema,
```
Comment (ricott1) [inline proposals/lip-0043.md:152]: Minor thing, but for consistency I would keep the `item` property in the last position (after `fieldNumber`)

File: proposals/lip-0043.md [hunk 4]
```diff
@@ -378,24 +365,10 @@ mainchainRegParams = {
         },
         "mainchainValidators": {
             "type": "array",
+            "items": ...blsKeyAndBftWeightSchema,
```
Comment (ricott1) [inline proposals/lip-0043.md:368]: same as above

**Review by ricott1 — CHANGES_REQUESTED** [review]
Just a minor comment to respect the previous order of the schema properties

File: proposals/lip-0043.md (hunk 4 above)
Comment (mjerkov) [reply proposals/lip-0043.md]: Noted.

File: proposals/lip-0043.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mjerkov) [reply proposals/lip-0043.md]: Please check now, should be fine.

**Review by janhack — APPROVED** [review]
Looks good.

**Review by ricott1 — APPROVED** [review]

**Review by janhack — APPROVED** [review]

**Pull request merged**